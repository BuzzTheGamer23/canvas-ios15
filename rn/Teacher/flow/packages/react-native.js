// @flow
export type PanResponderGestureState = {
  stateID: number,
  moveX: number,
  moveY: number,
  x0: number,
  y0: number,
  dx: number,
  dy: number,
  vx: number,
  vy: number,
}
