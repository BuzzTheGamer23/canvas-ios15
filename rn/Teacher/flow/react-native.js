//
// Copyright (C) 2017-present Instructure, Inc.
//
// This program is free software: you can redistribute it and/or modify
// it under the terms of the GNU General Public License as published by
// the Free Software Foundation, version 3 of the License.
//
// This program is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU General Public License for more details.
//
// You should have received a copy of the GNU General Public License
// along with this program.  If not, see <http://www.gnu.org/licenses/>.
//

// @flow
/* eslint-disable */

// define basic style prop types
// the internal ones are the best we have
import {
  type ____ViewStyleProp_Internal,
  type ____TextStyleProp_Internal,
  type ____ImageStyleProp_Internal,
} from 'react-native/Libraries/StyleSheet/StyleSheetTypes'

declare type ViewStyleProp = ____ViewStyleProp_Internal
declare type TextStyleProp = ____TextStyleProp_Internal
declare type ImageStyleProp = ____ImageStyleProp_Internal
