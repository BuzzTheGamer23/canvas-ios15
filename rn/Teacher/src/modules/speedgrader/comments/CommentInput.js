// @flow

import React, { Component } from 'react'
import {
  View,
  StyleSheet,
  Image,
  TextInput,
} from 'react-native'
import Button from 'react-native-button'
import i18n from 'format-message'
import Images from '../../../images'
import colors from '../../../common/colors'
import KeyboardSpacer from 'react-native-keyboard-spacer'
import DrawerState from '../utils/drawer-state'

type CommentInputProps = {
  makeComment(comment: SubmissionCommentParams): void,
  drawerState: DrawerState,
}

type State = {
  textComment: string,
}

export default class CommentInput extends Component<any, CommentInputProps, any> {
  state: State

  constructor (props: CommentInputProps) {
    super(props)
    this.state = { textComment: '' }
  }

  addMedia = () => {
    console.log('Add some media!')
  }

  submitComment = () => {
    this.props.makeComment({
      type: 'text',
      message: this.state.textComment,
    })
  }

  keyboardChanged = (visible: boolean) => {
    if (visible) {
      this.props.drawerState.snapTo(2, true)
    }
  }

  render () {
    const placeholder = i18n({
      default: 'Comment',
      description: 'Placeholder text for comment input field',
    })

    const addMedia = i18n({
      default: 'Add Media',
      description: 'Attach media to a message',
    })

    return (
      <View>
        <View style={styles.toolbar} >
          <Button
            containerStyle={styles.mediaButton}
            testID='submission-comment.add-media'
            onPress={this.addMedia}
            accessible
            accessibilityTraits={['button']}
            accessibilityLabel={addMedia}
          >
            <Image
              resizeMode="center"
              source={Images.add}
              style={styles.plus}
            />
          </Button>
          <View style={styles.inputContainer} >
            <TextInput
              autoFocus
              multiline
              testID='submission-comment.text-input'
              placeholder={placeholder}
              placeholderTextColor={colors.lightText}
              style={styles.input}
              maxHeight={76}
              onSubmitEditing={this.submitComment}
            />
          </View>
        </View>
        <KeyboardSpacer onToggle={this.keyboardChanged} />
      </View>
    )
  }
}

const styles = StyleSheet.create({
  mediaButton: {
    // paddingTop: 6,
    alignSelf: 'center',
  },
  plus: {
    tintColor: colors.secondaryButton,
  },
  toolbar: {
    overflow: 'hidden',
    backgroundColor: '#F5F5F5',
    borderTopWidth: StyleSheet.hairlineWidth,
    borderTopColor: colors.seperatorColor,
    flexDirection: 'row',
    alignItems: 'flex-start',
    justifyContent: 'space-between',
    paddingVertical: 8,
    paddingHorizontal: global.style.defaultPadding,
  },
  inputContainer: {
    flex: 1,
    borderRadius: 20,
    borderWidth: StyleSheet.hairlineWidth,
    borderColor: colors.seperatorColor,
    overflow: 'hidden',
    backgroundColor: 'white',
    marginLeft: 4,
  },
  input: {
    fontSize: 17,
    marginHorizontal: 10,
    marginTop: 0,
    marginBottom: 4,
  },
})
