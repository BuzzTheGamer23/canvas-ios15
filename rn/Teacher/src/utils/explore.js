/* @flow */

function hasProp (object: any, key: string, value: any): boolean {
  if (object instanceof Object) {
    const props = object.props
    if (props && props instanceof Object) {
      for (var prop in props) {
        if (prop === key && props[prop] === value) {
          return true
        }
      }
    }
  }

  return false
}

function selectByProp (object: any, propKey: string, value: any): ?any {
  if (!(typeof object === 'object')) {
    return null
  }

  if (hasProp(object, propKey, value)) {
    return object
  }

  let result = null

  for (var key in object) {
    if (object[key] instanceof Array) {
      object[key].forEach((e) => {
        result = selectByProp(e, propKey, value)
        if (result) {
          return result
        }
      })
    }

    result = selectByProp(object[key], propKey, value)
    if (result) {
      return result
    }
  }

  return result
}

class ComponentExplorer {
  component: any

  constructor (component: any) {
    this.component = component
  }

  selectByProp (propKey: string, value: any): ?any {
    return selectByProp(this.component, propKey, value)
  }

  selectByID (id: string): ?any {
    return this.selectByProp('testID', id)
  }
}

export default function explore (component: any): ComponentExplorer {
  return new ComponentExplorer(component)
}
