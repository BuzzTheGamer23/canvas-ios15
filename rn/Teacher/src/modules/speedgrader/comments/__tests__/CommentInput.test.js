// @flow

import 'react-native'
import React from 'react'
import renderer from 'react-test-renderer'
import CommentInput from '../CommentInput'
import explore from '../../../../../test/helpers/explore'

jest.mock('react-native-button', () => 'Button')

test('CommentInput renders', () => {
  let tree = renderer.create(
    <CommentInput makeComment={jest.fn()} />
  ).toJSON()
  expect(tree).toMatchSnapshot()
})

test('Wont render the media buttons when allowMediaInputs is set to false', () => {
  let tree = renderer.create(
    <CommentInput allowMediaComments={false} />
  ).toJSON()
  expect(tree).toMatchSnapshot()
})

test('Calls props.makeComment when the submit button is pressed', () => {
  let makeComment = jest.fn()
  let tree = renderer.create(
    <CommentInput makeComment={makeComment} />
  )

  tree.getInstance().setState({ textComment: 'text' })
  let button = explore(tree.toJSON()).selectByID('submit-comment') || {}
  button.props.onPress()

  expect(makeComment).toHaveBeenCalledWith({
    type: 'text',
    message: 'text',
  })
})
