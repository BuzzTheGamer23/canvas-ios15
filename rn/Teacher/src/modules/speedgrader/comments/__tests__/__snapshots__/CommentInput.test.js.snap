// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CommentInput renders 1`] = `
<View>
  <View
    style={
      Object {
        "alignItems": "flex-start",
        "backgroundColor": "#F5F5F5",
        "borderTopColor": "#C7CDD1",
        "borderTopWidth": 0.5,
        "flexDirection": "row",
        "justifyContent": "space-between",
        "overflow": "hidden",
        "paddingHorizontal": 16,
        "paddingVertical": 8,
      }
    }
  >
    <View
      accessibilityComponentType="button"
      accessibilityLabel={undefined}
      accessibilityTraits="button"
      accessible={true}
      hitSlop={undefined}
      isTVSelectable={true}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignSelf": "center",
          "opacity": 1,
        }
      }
      testID="submission-comment.add-media"
      tvParallaxProperties={undefined}
    >
      <Image
        resizeMode="center"
        source={1}
        style={
          Object {
            "tintColor": "#73818C",
          }
        }
      />
    </View>
    <View
      style={
        Object {
          "backgroundColor": "white",
          "borderColor": "#C7CDD1",
          "borderRadius": 20,
          "borderWidth": 0.5,
          "flex": 1,
          "marginLeft": 4,
          "overflow": "hidden",
        }
      }
    >
      <TextInput
        autoFocus={true}
        maxHeight={76}
        multiline={true}
        onSubmitEditing={[Function]}
        placeholder="Comment"
        placeholderTextColor="#7F91A7"
        style={
          Object {
            "fontSize": 17,
            "marginBottom": 4,
            "marginHorizontal": 10,
            "marginTop": 0,
          }
        }
        testID="submission-comment.text-input"
      />
    </View>
  </View>
  <View
    style={
      Array [
        Object {
          "bottom": 0,
          "left": 0,
          "right": 0,
        },
        Object {
          "height": 0,
        },
        undefined,
      ]
    }
  />
</View>
`;
